src/views/activity/activity_item_view.js

```diff
@@ -21,5 +27,35 @@ export default BaseView.extend({
 
   sendSelectionDetails (event) {
     webChannel.sendUrlSelected(this.model.url, 'url');
+  },
+
+  injectFavicon () {
+    // Manually construct the favicon element and inject it into the DOM,
+    // allowing us to handle events directly, since load events don't bubble
+    // and cannot be delegated.
+    this.icon = new Image('img');
+    this.icon.onload = this.calculateIconColor();
+    this.icon.onerror = this.hideErroredFavicon();
+    this.icon.src = this.model.displayImage();
+    this.iconContainer.appendChild(this.icon);
+  },
+
+  hideErroredFavicon () {
+    return () => {
+      this.icon.style.display = 'none';
+    }.bind(this);
+  },
+
+  calculateIconColor () {
+    // If able to do so in 20ms or less, calculate the dominant color of the
+    // favicon and set that to the background color of its parent.
+    return () => {
+      const START = new Date().getTime();
```

Comment:
I've always had the notion that APIs like `window.performance` are great for debugging, but not for any functionality-affecting production code, as this does. I couldn't find anything on the internet to support that notion, though.

I'm not sure `mark`/`measure` is the right call for this, as it seems to be intended for more global one-time events, rather than utility methods that might be run dozens of times in a page's lifetime. Unless I'm misunderstanding the API, we'd need to do something like:

```javascript
icon_count = 0;

calculateIconColor () {
  window.performance.mark('icon_color_start_' + icon_count);
  const favicon = new Favicon(this.icon);
  window.performance.mark('icon_color_end_' + icon_count);
  const elapsed = window.performance.measure('icon_color_start_' + icon_count,
                                             'icon_color_end_' + icon_count);
  ++icon_count;
}

```

That feels pretty bad to me. Thoughts?